src/cmap/connection.ts

```diff
@@ -366,6 +366,18 @@ export class Connection extends TypedEventEmitter<ConnectionEvents> {
         clusterTime = session.clusterTime;
       }
 
+      // We need to unpin any read or write commands that happen outside of a pinned
+      // transaction, so we check if we have a pinned transaction that is no longer
+      // active, and unpin for all except start or commit.
```

Comment:
Comment says it all here. :)

Reply:
I'm assuming the conditions excluding the start and commit are there because `transaction.isActive` is false in those cases?

Reply:
Yes, start is not active yet and commit could be any commit after the transaction is already committed, which we test for. (multiple commits are allowed)